projects/igniteui-angular/src/lib/date-picker/date-picker.utils.ts

```diff
@@ -0,0 +1,730 @@
+        }
+    } else {
+        yearPrefix = '20';
+    }
+    const fullYear = (yearFormat === FORMAT_DESC.TWO_DIGITS) ? yearPrefix.concat(year) : year;
+
+    if ((month < 0) || (month > 11)) {
+        return { state: DATE_STATE.INVALID, value: inputValue };
+    }
+
+    if ((day < 1) || (day > daysInMonth(fullYear, month + 1))) {
+        return { state: DATE_STATE.INVALID, value: inputValue };
+    }
+
+    return { state: DATE_STATE.VALID, date: new Date(fullYear, month, day) };
+}
+
+/**
+ *@hidden
+ */
+export function maskToPromptChars(mask: string): string {
+    return mask.replace(/0|L/g, PROMPT_CHAR);
+}
+
+/**
+ *@hidden
+ */
+export function trimUnderlines(value: string): string {
+    return value.replace(/_/g, '');
+}
+
+/**
+ *@hidden
+ */
+export function getNumericFormatPrefix(formatType: string): string {
+    switch (formatType) {
+        case FORMAT_DESC.TWO_DIGITS: {
+            return '0';
+        }
+        case FORMAT_DESC.NUMERIC: {
+            return PROMPT_CHAR;
+        }
+    }
+}
+
+/**
+ *@hidden
+ */
+export function getModifiedDateInput(dateFormatParts: any[],
+    inputValue: string,
+    position: number,
+    delta: number,
+    isSpinLoop: boolean): string {
+    const datePart = getDatePartOnPosition(dateFormatParts, position);
+    const datePartType = datePart.type;
+    const datePartFormatType = datePart.formatType;
+    let newValue;
+
+    const datePartValue = getDateValueFromInput(dateFormatParts, datePartType, inputValue);
+    newValue = parseInt(datePartValue, 10);
+
+    let maxValue, minValue;
+    const minMax = getMinMaxValue(dateFormatParts, datePart, inputValue);
+    minValue = minMax[0];
+    maxValue = minMax[1];
+
+    if (isNaN(newValue)) {
+        if (minValue === 'infinite') {
+            newValue = 2000;
+        } else {
+            newValue = minValue;
+        }
+    }
+    let tempValue = newValue;
+    tempValue += delta;
+
+    // Infinite loop for full years
+    if (maxValue === 'infinite' && minValue === 'infinite') {
+        newValue = tempValue;
+    }
+
+    if (isSpinLoop) {
+        if (tempValue > maxValue) {
+            tempValue = minValue;
+        }
+        if (tempValue < minValue) {
+            tempValue = maxValue;
+        }
+        newValue = tempValue;
+    } else {
+        if (tempValue <= maxValue && tempValue >= minValue) {
+            newValue = tempValue;
+        }
+    }
+
+    const startIdx = datePart.position[0];
+    const endIdx = datePart.position[1];
+    const start = inputValue.slice(0, startIdx);
+    const end = inputValue.slice(endIdx, inputValue.length);
+    let changedPart: string;
+
+    const prefix = getNumericFormatPrefix(datePartFormatType);
+    changedPart = (newValue < 10) ? `${prefix}${newValue}` : `${newValue}`;
+
+    return `${start}${changedPart}${end}`;
+}
+
+/**
+ *@hidden
+ */
+export function getMinMaxValue(dateFormatParts: any[], datePart, inputValue: string): any[] {
```

Comment:
This could be done by returning an object instead of an array. Something like:
`return { minValue: minValue, maxValue: maxValue };`

If this method is meant to be used publicly there should be an interface describing the returned value.